backend/src/main/java/gov/cdc/usds/simplereport/service/DeviceTypeService.java

```diff
@@ -42,6 +51,12 @@ public DeviceType getDeviceType(String internalId) {
         return _repo.findById(actualId).orElseThrow(()->new IllegalGraphqlArgumentException("invalid device type ID"));
     }
 
+    public DeviceSpecimen getDefaultForDeviceId(String deviceId) {
+        UUID actualDeviceId = UUID.fromString(deviceId);
+        return _deviceSpecimenRepo.findFirstByDeviceTypeInternalIdOrderByCreatedAt(actualDeviceId).orElseThrow(
```

Comment:
This is a backward-compatibility shim—I should mark it as such in javadocs.

The goal here is "given that I need a DeviceSpecimen, but I was given the ID for a DeviceType, what DeviceSpecimen should I assume that the caller wanted?" The logical flow to the answer is

* the earliest-created DeviceSpecimen for each DeviceType is the one we create during migrations
* if you are querying for a DeviceType directly, it's because you don't know that DeviceTypes can have multiple SpecimenTypes
* ergo, if you are just handing me a device type ID, I should look for the DeviceSpecimen that was created first, which will correspond to what the caller assumes is the only possible specimen collection type for this device

When we shift the front end to looking up DeviceSpecimen directly, this can go away.

In the short term, I could also have looked up the DeviceType and then found the relevant SpecimenType using the `typeCode`, but we probably will want to drop that column sooner than we want to drop support for the fallback API shims.

Reply:
well, this comment may be overly nitpicky because this is all temporary anyway, but i have the same concerns here about `findFirstByDeviceTypeInternalIdOrderByCreatedAt` not filtering out deleted `DeviceSpecimen`s